clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -3200,8 +3163,16 @@ class SyclKernelIntFooterCreator : public SyclKernelFieldHandler {
       : SyclKernelFieldHandler(S), Footer(F) {
     (void)Footer; // workaround for unused field warning
   }
-};
 
+  bool handleStructType(FieldDecl *, QualType FieldTy) final {
+    Footer.addTypeToCheckForDeviceCopyability(FieldTy);
+    return true;
+  }
+  bool handleArrayType(FieldDecl *FD, QualType FieldTy) final {
+    Footer.addTypeToCheckForDeviceCopyability(FieldTy);
```

Comment:
This line I think is wrong, the previous checks grabbed the element type of the array for the check.